app/controllers/admin/users_controller.rb

```diff
@@ -0,0 +1,89 @@
+class Admin::UsersController < ApplicationController
+  before_action :set_user, only: %i[ show edit update destroy ]
+  helper_method :sort_column, :sort_direction
+  # GET /users or /users.json
+  def index
+    if params[:query].present?
+      @pagy, @users = pagy(User.search(params[:query]), items: 2)
+    elsif sort_column and sort_direction
+      @pagy, @users = pagy(User.order(sort_column + ' ' + sort_direction), items: 3)
+    else
+      @pagy, @users = pagy(User.all, items: 3)
+    end
```

Comment:
let's move these to model, create a class method and pass required controller params as parameters to model

Reply:
I cannot move these to model, bilal bhai. I have two users_controller. One is in the admin folder  so if I move it to model then it will be applied to the other users_controller too but I only need this in admin/users